File: src/main/java/com/wikia/tibia/repositories/MissileRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.MissileGateway;
import com.wikia.tibia.objects.Missile;

public class MissileRepository extends WikiObjectRepository {

    public MissileRepository() {
        super(Missile.class, new MissileGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/MissileRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Missile

class MissileRepository : WikiObjectRepository<Missile>(
    wikiObjectClass = Missile::class.java,
    wikiObjectGateway = WikiObjectGateway(Contract.MISSILES)
)
